lib/bloc/webhooks/webhooks_bloc.dart

```diff
@@ -0,0 +1,185 @@
+      _log.info("SDK webhook registered: $webhookUrl");
+      final lnurl = await registerLnurlpay(nodeState, webhookUrl);
+      emit(state.copyWith(lnurlpayUrl: lnurl));
+    } catch (err) {
+      _log.warning("Failed to register webhooks: $err");
+      emit(state.copyWith(lnurlPayError: err.toString()));
+    }
+  }
+
+  Future<String> registerLnurlpay(NodeState nodeState, String webhookUrl) async {
+    final sharedPrefs = await injector.sharedPreferences;
+    var lastUsedLnurlpay = sharedPrefs.getString("lnurlpay_key");
+    if (lastUsedLnurlpay != null && lastUsedLnurlpay != webhookUrl) {
+      await _invalidateLnurlpay(nodeState, lastUsedLnurlpay);
+    }
+    final currentTime = DateTime.now().millisecondsSinceEpoch ~/ 1000;
+    final signature =
+        await _breezSDK.signMessage(req: SignMessageRequest(message: "$currentTime-$webhookUrl"));
+    String lnurlWebhookUrl = "$lnurlServiceURL/lnurlpay/${nodeState.id}";
+    final uri = Uri.parse(lnurlWebhookUrl);
+    final jsonResponse = await http.post(uri,
+        body: jsonEncode(AddWebhookRequest(
+          time: currentTime,
+          webhookUrl: webhookUrl,
+          signature: signature.signature,
+        ).toJson()));
+    if (jsonResponse.statusCode == 200) {
+      final data = jsonDecode(jsonResponse.body);
+      final lnurl = data['lnurl'];
+      _log.info("lnurlpay webhook registered: $webhookUrl, lnurl = $lnurl");
+      await sharedPrefs.setString("lnurlpay_key", webhookUrl);
+      return lnurl;
+    } else {
+      throw jsonResponse.body;
+    }
+  }
+
+  Future _invalidateLnurlpay(NodeState nodeState, String toInvalidate) async {
+    final sharedPrefs = await injector.sharedPreferences;
+    String lnurlWebhookUrl = "$lnurlServiceURL/lnurlpay/${nodeState.id}";
+    final currentTime = DateTime.now().millisecondsSinceEpoch ~/ 1000;
+    final signature =
+        await _breezSDK.signMessage(req: SignMessageRequest(message: "$currentTime-$toInvalidate"));
+    final uri = Uri.parse(lnurlWebhookUrl);
+    final response = await http.delete(uri,
+        body: jsonEncode(RemoveWebhookRequest(
+          time: currentTime,
+          webhookUrl: toInvalidate,
+          signature: signature.signature,
+        ).toJson()));
+    _log.info("invalidate lnurl pay response: ${response.statusCode}");
+    await sharedPrefs.remove("lnurlpay_key");
+  }
+
+  Future<String> generateWebhookURL() async {
+    final notifications = injector.notifications;
+    String? token = await notifications.getToken();
+
+    _log.info("Retrieved token, registering…");
+
+    String platform = defaultTargetPlatform == TargetPlatform.iOS
+        ? "ios"
+        : defaultTargetPlatform == TargetPlatform.android
+            ? "android"
+            : "";
+    if (platform.isEmpty) {
+      throw Exception("Notifications for platform is not supported");
+    }
+    return "$notifierServiceURL/api/v1/notify?platform=$platform&token=$token";
+  }
+
+  String generateUser() {
```

Comment:
This function is never used and it's better off being an utility function under utils folder.

Reply:
Right, sneaked by mistake.